docs/debugging_guide.md

```diff
@@ -0,0 +1,362 @@
+---
+layout: spec
+title: Debugging Guide
+excludeFromSitemap: true
+---
+
+Debugging Guide
+==========================
+{: .primer-spec-toc-ignore }
+
+There are two fundamental components to debugging:
+1. Form hypotheses about the specific cause of a bug
+2. Use debugging tools and strategies to investigate those hypotheses
+
+This guide focuses on effective strategies for point 2 above, in particular through the use of a **debugger**, which allows you to pause your program's execution at key points and inspect the state of objects in memory.
+
+We also show several real examples drawn from course projects in EECS 280. In some of the cases, we've blurred out parts of screenshots where we ran the debugger on our instructor solution.
+
+TODO: link back to relevant parts of IDE or setup tutorials for debuggers
+
+## Inspect Program State
+
+A fundamental part of debugging is simply getting more information about what exactly your program is doing.
+
+### Local Variables
+
+<div class="primer-spec-callout info icon-info" markdown="1">
+Use the **variables panel** to inspect the values of variables while your debugger is paused.
```

Comment:
Do you think it's worth mentioning that you can hover the mouse pointer over a variable in the source code?

Does the variables pane advice apply to the other debuggers?  I know that the hover does.

Reply:
Good call! I think this is worth a mention. Added in de6d5b4

Reply:
Good call! I think this is worth a mention. Added in de6d5b4

Reply:
I believe all the variables panel advice applies to the three main IDEs/editors we recommend.